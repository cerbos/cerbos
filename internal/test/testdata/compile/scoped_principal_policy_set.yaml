# yaml-language-server: $schema=../.jsonschema/CompileTestCase.schema.json
---
mainDef: "donald_duck.acme.hr.uk.yaml"
