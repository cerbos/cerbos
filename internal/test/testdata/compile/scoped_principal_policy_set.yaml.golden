{
  "fqn": "cerbos.principal.donald_duck.v20210210/acme.hr.uk",
  "principalPolicy": {
    "meta": {
      "fqn": "cerbos.principal.donald_duck.v20210210/acme.hr.uk",
      "principal": "donald_duck",
      "version": "20210210"
    },
    "policies": [
      {
        "scope": "acme.hr.uk",
        "resourceRules": {
          "leave_request": {
            "actionRules": [
              {
                "action": "*",
                "name": "leave_request_rule-001",
                "effect": "EFFECT_DENY"
              }
            ]
          }
        }
      },
      {
        "scope": "acme.hr",
        "resourceRules": {
          "leave_request": {
            "actionRules": [
              {
                "action": "*",
                "name": "leave_request_rule-001",
                "effect": "EFFECT_ALLOW"
              }
            ]
          }
        }
      },
      {
        "scope": "acme",
        "resourceRules": {
          "leave_request": {
            "actionRules": [
              {
                "action": "*",
                "name": "leave_request_rule-001",
                "condition": {
                  "expr": {
                    "original": "request.resource.attr.dev_record == true",
                    "checked": {
                      "referenceMap": {
                        "1": {
                          "name": "request"
                        },
                        "5": {
                          "overloadId": [
                            "equals"
                          ]
                        }
                      },
                      "typeMap": {
                        "1": {
                          "messageType": "cerbos.engine.v1.Request"
                        },
                        "2": {
                          "messageType": "cerbos.engine.v1.Request.Resource"
                        },
                        "3": {
                          "mapType": {
                            "keyType": {
                              "primitive": "STRING"
                            },
                            "valueType": {
                              "dyn": {}
                            }
                          }
                        },
                        "4": {
                          "dyn": {}
                        },
                        "5": {
                          "primitive": "BOOL"
                        },
                        "6": {
                          "primitive": "BOOL"
                        }
                      },
                      "sourceInfo": {
                        "location": "<input>",
                        "lineOffsets": [
                          41
                        ],
                        "positions": {
                          "1": 0,
                          "2": 7,
                          "3": 16,
                          "4": 21,
                          "5": 33,
                          "6": 36
                        }
                      },
                      "expr": {
                        "id": "5",
                        "callExpr": {
                          "function": "_==_",
                          "args": [
                            {
                              "id": "4",
                              "selectExpr": {
                                "operand": {
                                  "id": "3",
                                  "selectExpr": {
                                    "operand": {
                                      "id": "2",
                                      "selectExpr": {
                                        "operand": {
                                          "id": "1",
                                          "identExpr": {
                                            "name": "request"
                                          }
                                        },
                                        "field": "resource"
                                      }
                                    },
                                    "field": "attr"
                                  }
                                },
                                "field": "dev_record"
                              }
                            },
                            {
                              "id": "6",
                              "constExpr": {
                                "boolValue": true
                              }
                            }
                          ]
                        }
                      }
                    }
                  }
                },
                "effect": "EFFECT_ALLOW"
              }
            ]
          },
          "salary_record": {
            "actionRules": [
              {
                "action": "*",
                "name": "salary_record_rule-001",
                "effect": "EFFECT_DENY"
              }
            ]
          }
        }
      },
      {
        "resourceRules": {
          "leave_request": {
            "actionRules": [
              {
                "action": "*",
                "name": "leave_request_rule-001",
                "effect": "EFFECT_ALLOW"
              }
            ]
          }
        }
      }
    ]
  },
  "compilerVersion": 1
}
